boofuzz/primitives/bit_field.py

```diff
@@ -106,34 +107,59 @@ def __init__(
                     self._fuzz_library.append(val)
             else:
                 # try only "smart" values.
-                self.add_integer_boundaries(0)
-                self.add_integer_boundaries(self.max_num // 2)
-                self.add_integer_boundaries(self.max_num // 3)
-                self.add_integer_boundaries(self.max_num // 4)
-                self.add_integer_boundaries(self.max_num // 8)
-                self.add_integer_boundaries(self.max_num // 16)
-                self.add_integer_boundaries(self.max_num // 32)
-                self.add_integer_boundaries(self.max_num)
+                self.add_integer_boundaries(0, 10)
+                self.add_integer_boundaries(self.max_num // 2, 10)
+                self.add_integer_boundaries(self.max_num // 3, 10)
+                self.add_integer_boundaries(self.max_num // 4, 10)
+                self.add_integer_boundaries(self.max_num // 8, 10)
+                self.add_integer_boundaries(self.max_num // 16, 10)
+                self.add_integer_boundaries(self.max_num // 32, 10)
+                self.add_integer_boundaries(self.max_num, 10)
+
+                power_of_two = math.log(self.max_num, 2)
+                bits = width if not self.max_num else int(math.floor(power_of_two))
```

Comment:
`power_of_two = math.log(self.max_num, 2)` will fail if  `self.max_num` is `None`.  Make this conditional situation more robust.

Also, for `bits = width if not self.max_num else int(math.floor(power_of_two))`, I'm not clear why the existence of max_num means you should use width.

Breaking this out into a proper if/then statement might make the whole thing look cleaner.